pyfaidx/__init__.py

```diff
@@ -997,15 +997,13 @@ def __init__(self,
             sequence_always_upper=sequence_always_upper,
             rebuild=rebuild,
             build_index=build_index)
-        self.keys = self.faidx.index.keys
-        self.values = self.faidx.index.values
-        self.items = self.faidx.index.items
         if not self.mutable:
-            self.records = dict(
-                [(rname, FastaRecord(rname, self)) for rname in self.keys()])
+            self.records = {rname: FastaRecord(rname, self)
```

Comment:
Great! You can ignore my previous comment.

Reply:
It seems like dictionary key insertion order is the issue, and I will update here to do something similar to what the `Fasta.keys` method currently relies on:

https://github.com/mdshw5/pyfaidx/blob/f1668d2d0fde95892519eaa78a33bc1891e310cc/pyfaidx/__init__.py#L399